app/src/sharedTest/java/org/oppia/android/app/customview/LessonThumbnailImageViewTest.kt

```diff
@@ -0,0 +1,179 @@
+
+    activityTestRule.runOnUiThread {
+      with(lessonThumbnailImageView) {
+        lessonThumbnailImageViewHolder.addView(this)
+        setEntityId("")
+        setEntityType("")
+        setLessonThumbnail(LessonThumbnail.getDefaultInstance())
+
+        verify(imageLoader, atLeastOnce()).loadDrawable(
+          imageDrawableResId = R.drawable.topic_fractions_01,
+          target = ImageViewTarget(this),
+          transformations = listOf()
+        )
+      }
+    }
+  }
+
+  // TODO(#59): Figure out a way to reuse modules instead of needing to re-declare them.
+  @Singleton
+  @Component(
+    modules = [
+      RobolectricModule::class,
+      TestDispatcherModule::class, ApplicationModule::class,
+      LoggerModule::class, ContinueModule::class, FractionInputModule::class,
+      ItemSelectionInputModule::class, MultipleChoiceInputModule::class,
+      NumberWithUnitsRuleModule::class, NumericInputRuleModule::class, TextInputRuleModule::class,
+      DragDropSortInputModule::class, ImageClickInputModule::class, InteractionsModule::class,
+      GcsResourceModule::class, TestModule::class, ImageParsingModule::class,
+      HtmlParserEntityTypeModule::class, QuestionModule::class, TestLogReportingModule::class,
+      AccessibilityTestModule::class, LogStorageModule::class, CachingTestModule::class,
+      PrimeTopicAssetsControllerModule::class, ExpirationMetaDataRetrieverModule::class,
+      ViewBindingShimModule::class, RatioInputModule::class,
+      ApplicationStartupListenerModule::class, LogUploadWorkerModule::class,
+      WorkManagerConfigurationModule::class, HintsAndSolutionConfigModule::class,
+      FirebaseLogUploaderModule::class, FakeOppiaClockModule::class
+    ]
+  )
+  interface TestApplicationComponent : ApplicationComponent {
+    @Component.Builder
+    interface Builder : ApplicationComponent.Builder
+
+    fun inject(lessonThumbnailImageViewTest: LessonThumbnailImageViewTest)
+  }
+
+  class TestApplication : Application(), ActivityComponentFactory, ApplicationInjectorProvider {
+    private val component: TestApplicationComponent by lazy {
+      DaggerLessonThumbnailImageViewTest_TestApplicationComponent.builder()
+        .setApplication(this)
+        .build() as TestApplicationComponent
+    }
+
+    fun inject(lessonThumbnailImageViewTest: LessonThumbnailImageViewTest) {
+      component.inject(lessonThumbnailImageViewTest)
+    }
+
+    override fun createActivityComponent(activity: AppCompatActivity): ActivityComponent {
+      return component.getActivityComponentBuilderProvider().get().setActivity(activity).build()
+    }
+
+    override fun getApplicationInjector(): ApplicationInjector = component
+  }
+
+  /** Provides test dependencies (including a mock for [ImageLoader] to capture its operations). */
+  @Module
+  class TestModule {
+    @Provides
+    @Singleton
+    fun provideMockImageLoader() = Mockito.mock(ImageLoader::class.java)
```

Comment:
I actually suggest using the changes from #2981: https://github.com/oppia/oppia-android/pull/2981/files#diff-90aff075f37a890b5a50c2e1bd541f4f357fd29185d65c2495a4ff22648f7a1bR68. I'll be merging that later tonight my time so it should be available on develop pretty soon. You can avoid mocking ImageLoader & instead use ``TestImageLoaderModule``. It makes things more consistent across the codebase.

Reply:
I used the updated TestGlideImageLoader and the withDrawable() Matcher. 
Robolectric tests are working fine (i.e. failing before fix and passing after fix) but the Expresso tests keep on running indefinitely. 
Should I push the updated test file for reference?

Reply:
Ah, looks like you didn't even need #2981.

I suspect you'll need to add a block like this one to get Glide set up correctly on Espresso:

https://github.com/oppia/oppia-android/blob/9782e2a526a55763d09786f821e739fac171e77a/app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt#L179-L190

Reply:
Just a small doubt. Are we using Glide here? I have used the TestImageLoaderModule for the test instead of GlideImageLoaderModule. And the loadDrawable() function of TestGlideImageLoader doesn't seem to be using Glide.

https://github.com/oppia/oppia-android/blob/9782e2a526a55763d09786f821e739fac171e77a/utility/src/main/java/org/oppia/android/util/parser/TestGlideImageLoader.kt#L48-L62

Reply:
Yes, the test image loader still proxies back to Glide itself for loading (we intentionally don't mock Glide out if we can help it since it's otherwise a bit difficult to make sure our integration with Glide is working correctly).

Reply:
I added the code block you linked. The problem still persists. Expresso tests keep on running indefinitely.
Should I push the changes?